pulp_rpm/app/tasks/publishing.py

```diff
@@ -408,6 +409,16 @@ def create_repomd_xml(
     pkg_to_hash = {}
     for ca in contentartifact_qs.iterator():
         pkgid = None
+        if (
+            package_checksum_type
+            and package_checksum_type not in settings.ALLOWED_CONTENT_CHECKSUMS
```

Comment:
This condition can be moved up, outside of the for loop, so this check doesn't happen for every artifact.
I believe package_checksum_type is a configuration on a repo or a publication and can be checked once.

Reply:
It will also benefit from a more specific error message saying what to reconfigure or what to do to mitigate that.